cfme/intelligence/chargeback/rates.py

```diff
@@ -224,19 +207,65 @@ def delete(self, cancel=False):
         """
         view = navigate_to(self, 'Details', wait_for_view=True)
         view.toolbar.configuration.item_select('Remove from the VMDB', handle_alert=(not cancel))
+        view = self.create_view(navigator.get_class(self.parent, 'All').VIEW)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+
+
+@attr.s
+class ComputeRateCollection(BaseCollection):
+    ENTITY = ComputeRate
+    RATE_TYPE = 'Compute'
+
+    def create(self, description=None, currency=None, fields=None):
+        # Create a rate in UI
+        rate = self.instantiate(description, currency, fields)
+
+        view = navigate_to(self, 'Add')
+        view.fill_with({'description': rate.description,
+                        'currency': rate.currency,
+                        'fields': rate.fields},
+                       on_change=view.add_button,
+                       no_change=view.cancel_button)
+
         view = self.create_view(navigator.get_class(self, 'All').VIEW)
         assert view.is_displayed
         view.flash.assert_no_error()
 
+        return rate
+
 
+@attr.s
 class StorageRate(ComputeRate):
     # Methods and form for this are similar to that of ComputeRate, but navigation is different
     # from that of ComputeRate.
     pretty_attrs = ['description']
     RATE_TYPE = 'Storage'
 
 
-@navigator.register(ComputeRate, 'All')
+class StorageRateCollection(BaseCollection):
+    ENTITY = StorageRate
+    RATE_TYPE = 'Storage'
+
```

Comment:
Please define all() methods if you can within scope. I need to discuss with other reviewers if implementations of all() are required.

Reply:
@mshriver I couldn't find an implementation of all() outside of ParameterizedView. Is it a Collection method to return a list of all Entity objects on the appliance?

Reply:
@LandoCalrizzian that's exactly it - instantiate list of entity objects from UI/REST/DB source of entity attributes (names, providers, description, etc).

Here's an example from the containers code, using the DB to query for volumes:
https://github.com/ManageIQ/integration_tests/blob/73b1aaba74cd5802774881022ecda34a404a8025/cfme/containers/volume.py#L46

And here's a simpler example from the network/security_group code, using the UI:
https://github.com/ManageIQ/integration_tests/blob/b3572621f7a0eba24826ae9d6cd0d6c470a0606d/cfme/networks/security_group.py#L51